strings/expressionConverter.cpp

```diff
@@ -0,0 +1,185 @@
+#include <iostream>   /// for IO operations
+#include <stack>      /// for std::stack
+#include <string>     /// for std::string
+
+/**
+ * @namespace expression
+ * @brief Algorithms for expression parsing and conversion
+ */
+namespace expression {
+
+/**
+ * @brief Checks if the given character is a supported operator.
+ * @param c input character
+ * @return true if operator, false otherwise
+ */
+bool is_operator(char c) {
+    return c == '+' || c == '-' || c == '*' || c == '/' || c == '^';
+}
+
+/**
+ * @brief Returns precedence level of operator.
+ * @param op operator character
+ * @return integer precedence
+ */
+int precedence(char op) {
+    if (op == '^')
+        return 3;
+    if (op == '*' || op == '/')
+        return 2;
+    if (op == '+' || op == '-')
+        return 1;
+    return 0;
+}
+
+/**
+ * @brief Returns true if operator is right-associative.
+ * @param op operator character
+ * @return true if right-associative
+ */
+bool is_right_associative(char op) { return op == '^'; }
+
+/**
+ * @brief Converts an infix expression to postfix notation.
+ * @param expr input infix expression
+ * @return postfix expression as string
+ */
+std::string infix_to_postfix(const std::string &expr) {
+    std::stack<char> ops;
+    std::string result;
+
+    for (char token : expr) {
+        if (std::isspace(token))
+            continue;
+
+        if (std::isalnum(token)) {
+            result += token;
+        } else if (token == '(') {
+            ops.push(token);
+        } else if (token == ')') {
+            while (!ops.empty() && ops.top() != '(') {
+                result += ops.top();
+                ops.pop();
+            }
+            if (!ops.empty())
+                ops.pop();  // discard '('
+        } else if (is_operator(token)) {
+            while (!ops.empty() && is_operator(ops.top())) {
+                char top = ops.top();
+                if ((is_right_associative(token) &&
+                     precedence(token) < precedence(top)) ||
+                    (!is_right_associative(token) &&
+                     precedence(token) <= precedence(top))) {
+                    result += top;
+                    ops.pop();
+                } else {
+                    break;
+                }
+            }
+            ops.push(token);
+        }
+    }
+
+    while (!ops.empty()) {
+        result += ops.top();
+        ops.pop();
+    }
+
+    return result;
+}
+
+/**
+ * @brief Converts an infix expression to prefix notation.
+ * @param expr input infix expression
+ * @return prefix expression as string
+ */
+std::string infix_to_prefix(const std::string &expr) {
+    std::string rev_expr = expr;
+    std::reverse(rev_expr.begin(), rev_expr.end());
+
+    for (char &ch : rev_expr) {
+        if (ch == '(')
+            ch = ')';
+        else if (ch == ')')
+            ch = '(';
+    }
+
+    std::string postfix = infix_to_postfix(rev_expr);
+    std::reverse(postfix.begin(), postfix.end());
```

Comment:
Is there no way to implement the prefix directly?

This repository is for learning so It may be better to include the implementation.